chrome-habit-tracker-master/index.js

```diff
@@ -0,0 +1,222 @@
+'use strict';
+
+//Declare global variables
+let habitStorage = {'habits': []}; //For updating sync storage
+let actualDays; //Days in selected month for rendering
+let monthSelected = new Date().getMonth(); //Actual month selected 0-indexed
+
+document.addEventListener('DOMContentLoaded', () => {
+  chrome.storage.sync.get('habits', (habits) => {
+    //Check sync storage
+    if (habits.hasOwnProperty('habits')) Object.assign(habitStorage, habits);
+    //Chrome doesn't allow this to be put directly on index.html
+    document.getElementById('add-habit-button').addEventListener('click', showInput);
```

Comment:
when adding listeners most browsers  remove listeners themselves . There are some older browsers (IE 6 and 7, if i recall correctly) that are messing this up. but there could be memory leaks. You should not have to worry about this, but you need to worry about removing event listeners